Code/Tools/TestImpactFramework/Runtime/Python/Code/Source/TestEngine/Python/TestImpactPythonTestEngine.cpp

```diff
@@ -71,6 +104,51 @@ namespace TestImpact
         DeleteFiles(m_artifactDir.m_coverageArtifactDirectory, "*.pycoverage");
     }
 
+    TestEngineRegularRunResult<PythonTestTarget> PythonTestEngine::RegularRun(
+        const AZStd::vector<const PythonTestTarget*>& testTargets,
+        Policy::ExecutionFailure executionFailurePolicy,
+        Policy::TestFailure testFailurePolicy,
+        Policy::TargetOutputCapture targetOutputCapture,
+        AZStd::optional<AZStd::chrono::milliseconds> testTargetTimeout,
+        AZStd::optional<AZStd::chrono::milliseconds> globalTimeout,
+        AZStd::optional<TestEngineJobCompleteCallback<PythonTestTarget>> callback) const
+    {
+        DeleteArtifactXmls();
+
+        const auto jobInfos = m_regularTestJobInfoGenerator->GenerateJobInfos(testTargets);
+
+        if (m_testRunnerPolicy == Policy::TestRunner::UseNullTestRunner)
+        {
+            return RunTests(
+                m_regularNullTestRunner.get(),
+                jobInfos,
+                testTargets,
+                PythonRegularTestRunnerErrorCodeChecker,
+                executionFailurePolicy,
+                testFailurePolicy,
+                targetOutputCapture,
+                testTargetTimeout,
+                globalTimeout,
+                callback,
+                AZStd::nullopt);
+        }
+        else
+        {
+            return RunTests(
+                m_regularTestRunner.get(),
+                jobInfos,
+                testTargets,
+                PythonRegularTestRunnerErrorCodeChecker,
+                executionFailurePolicy,
+                testFailurePolicy,
```

Comment:
Could this just be reduced to....

```
return RunTests(
    m_testRunnerPolicy == Policy::TestRunner::UseNullTestRunner
      ? m_regularNullTestRunner.get()
      : m_regularTestRunner.get(),
    jobInfos,
    testTargets,
    PythonRegularTestRunnerErrorCodeChecker,
    executionFailurePolicy,
    testFailurePolicy,
    targetOutputCapture,
    testTargetTimeout,
    globalTimeout,
    callback,
    AZStd::nullopt);
```

Reply:
They're different types so that won't work, unfortunately. I'll dig further for possible solutions in a followup PR 👍